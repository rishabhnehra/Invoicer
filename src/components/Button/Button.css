.button {
    font-size: 15px;
    color: #FFF;
    width: 200px;
    height: 42px;
    border-radius: 5px;
    border: none;
    box-shadow: 0px 3px 6px rgba(0,0,0,0.4);
    box-sizing: border-box;
}

.primary {
    background-color: #9E579D;
}

.secondary {
    background-color: #574B90;
}

.accent {
    background-color: #FC85AE;
}

.circular {
    border-radius: 20px;
}

.large {
    height: 49px;
}

.outline {
    color: #000;
    background-color: rgba(0,0,0,0);
    border: 3px solid #9E579D;
}
