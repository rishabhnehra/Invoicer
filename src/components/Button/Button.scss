.button {
    font-size: 15px;
    color: #FFF;
    padding: 0 22px;
    height: 42px;
    border-radius: 5px;
    border: none;
    box-shadow: 0px 3px 6px rgba(0,0,0,0.4);

    &.primary {
        background-color: #574B90; 
    }

    &.secondary {
        background-color:#9E579D;
    }

    &.accent {
        background-color: #FC85AE;
    }
    
    &.circular {
        border-radius: 20px;
    }

    &.large {
        height: 49px;
    }
    
    &.outline {
        color: #000;
        background-color: rgba(0,0,0,0);
        border: 3px solid #9E579D;
    }
}