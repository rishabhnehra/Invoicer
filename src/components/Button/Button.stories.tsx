import { Story, Meta } from '@storybook/react/types-6-0';

import { Button, ButtonProps, Colors } from './Button';

export default {
    title: 'Button',
    component: Button
} as Meta;

const Template: Story<ButtonProps> = (args) => <Button {...args}>This is button</Button>;

export const Primary = Template.bind({});
Primary.args = {
    color: Colors.PRIMARY
}

export const Secondary = Template.bind({});
Secondary.args = {
    color: Colors.SECONDARY
}

export const Accent = Template.bind({});
Accent.args = {
    color: Colors.ACCENT
}