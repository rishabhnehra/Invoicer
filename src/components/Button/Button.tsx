import './Button.css';

export enum Colors {
    Primary = "primary",
    Secondary = "secondary",
    Accent = "accent"
}

export type ButtonProps = {
    circular?: boolean,
    color?: Colors,
    customColor?: string,
    large?: boolean,
    outline?: boolean,
    onClick?: () => void
}

export const Button: React.FC<ButtonProps> = ({ 
    circular = false,
    color = Colors.Primary,
    large = false,
    outline = false,
    customColor,
    onClick,
    children 
}) => {
    const buttonClass = ['button', color];

    if (circular)
        buttonClass.push('circular');

    if (large)
        buttonClass.push('large');

    if (outline)
        buttonClass.push('outline');

    return <button
        className={buttonClass.join(' ')}
        onClick={onClick}
        style={{ backgroundColor: customColor }}
    >{children}</button>
}
