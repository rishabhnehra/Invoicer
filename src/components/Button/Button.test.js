import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Button } from './Button';

describe("<Button />", () => {
    it("snapshot check", () => {
        const tree = renderer.create(<Button>From Snapshot</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("should render button with \"This is buttton\" text", () => {
        render(<Button>This is button</Button>);
        expect(screen.getByText('This is button')).toBeInTheDocument();
    });
});