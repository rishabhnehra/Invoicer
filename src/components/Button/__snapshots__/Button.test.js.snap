// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> snapshot check 1`] = `
<button
  className="button primary"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  From Snapshot
</button>
`;
